docs/design/podaffinity.md

```diff
@@ -0,0 +1,615 @@
+systems. We believe that it properly balances the goal of expressiveness against the goals of
+simplicity and efficiency of implementation. However, we recognize that
+use cases may arise in the future that cannot be expressed using the syntax described here.
+Although we are not implementing an affinity-specific extensibility mechanism for a variety
+of reasons (simplicity of the codebase, simplicity of cluster deployment, desire for Kubernetes
+users to get a consistent experience, etc.), the regular Kubernetes
+annotation mechanism can be used to add or replace affinity rules. The way this work would is
+1. Define one or more annotations to describe the new affinity rule(s)
+1. User (or an admission controller) attaches the annotation(s) to pods to request the desired scheduling behavior.
+If the new rule(s) *replace* one or more fields of `Affinity` then the user would omit those fields
+from `Affinity`; if they are *additional rules*, then the user would fill in `Affinity` as well as the
+annotation(s).
+1. Scheduler takes the annotation(s) into account when scheduling.
+
+If some particular new syntax becomes popular, we would consider upstreaming it by integrating
+it into the standard `Affinity`.
+
+## Future work and non-work
+
+One can imagine that in the anti-affinity RequiredDuringScheduling case
+one might want to associate a number with the rule,
+for example "do not allow this pod to share a rack with more than three other
+pods (in total, or from the same service as the pod)." We could allow this to be
+specified by adding an integer `Limit` to `PodAffinityTerm` just for the
+`RequiredDuringScheduling` case. However, this flexibility complicates the
+system and we do not intend to implement it.
+
+It is likely that the specification and implementation of pod anti-affinity
+can be unified with [taints and tolerations](https://github.com/kubernetes/kubernetes/pull/18263),
+and likewise that the specification and implementation of pod affinity
+can be unified with [node affinity](https://github.com/kubernetes/kubernetes/pull/18261).
+The basic idea is that pod labels would be "inherited" by the node, and pods
+would only be able to specify affinity and anti-affinity for a node's labels.
+Our main motivation for not unifying taints and tolerations with
+pod anti-affinity is that we foresee taints and tolerations as being a concept that
+only cluster administrators need to understand (and indeed in some setups taints and
+tolerations wouldn't even be directly manipulated by a cluster administrator,
+instead they would only be set by an admission controller that is implementing the administrator's
+high-level policy about different classes of special machines and the users who belong to the groups
+allowed to access them). Moreover, the concept of nodes "inheriting" labels
+from pods seems complicated; it seems conceptually simpler to separate rules involving
```

Comment:
One could view this similarly to pod anti-affinity. The taints don't need to be actually propagated to nodes in order to be respected.